src/gtk/window.cpp

```diff
@@ -2870,6 +3343,81 @@ void wxWindowGTK::ConnectWidget( GtkWidget *widget )
                       G_CALLBACK (gtk_window_enter_callback), this);
     g_signal_connect (widget, "leave_notify_event",
                       G_CALLBACK (gtk_window_leave_callback), this);
+
+#if GTK_CHECK_VERSION(3,14,0)
+    if ( gtk_check_version(3, 14, 0) == NULL )
+    {
+        GtkGesture* vertical_pan_gesture = gtk_gesture_pan_new(widget, GTK_ORIENTATION_VERTICAL);
```

Comment:
Unfortunately I'm almost sure that this is a memory leak and this object (and all the other gestures) doesn't seem to be unref'd anywhere. This is a real problem because we just can't allow to leak 5 objects per each window created (and destroyed) by the program, this will very quickly result in big problems in real usage.

Unless I'm missing something -- but I just don't see how and when can this be destroyed currently...

Reply:
Catalin too pointed this out during GSoC, at that time I created a valgrind log and there weren't any big changes in memory leaks with or without gesture objects but that could also be resulted due to the fact that the log was created for a very program. Anyways, it's better to unref them and make sure no leaks happen. I'll give it a try.

Reply:
@prashantkn94 I'm working on this code right now and will try to push my putative changes out soon, so please wait for a moment -- but I'd be very grateful if you could help with testing them once I do push them out.